docs/RFCS/version_migration.md

```diff
@@ -0,0 +1,155 @@
+into the build. A node's version will be supplied with every RPC
+(e.g. Raft, Node, DistSQL, and Gossip). In addition, there will be a
+cluster-wide minimum required version, set by the operator. Each node
+also sends the minimum required version its operating under with each
+RPC.
+
+- RPCs from nodes with versions older than the minimum required
+  version will fail with an error that indicates the sender should
+  panic.
+- If an RPC is received which specifies a minimum required version
+  which is newer than the receiver's minimum required version, an
+  error is returned that indicates the sender should backoff and retry
+  (while the receiver waits for new gossip).
+- If an RPC is received which specifies a minimum required version
+  which is newer than the receiver's version, the receiver returns
+  an error and panics.
+
+The minimum required version will be stored and gossiped as a cluster
+setting. The value of the minimum required version is stored in a
+`ClusterVersion` protobuf, which to start will include just the
+minimum required version number.
+
+```
+message ClusterVersion {
+    int minimum_required_version = 1 [(gogoproto.nullable) = false];
+}
+```
+
+Nodes will listen for changes to the gossiped `ClusterVersion` value
+and panic with an error in the event that the
+`minimum_required_version` is greater than the their version.
+
+To see how this works, consider the following scenario involving a
+non-live node that resurfaces with adverse timing. In this scenario,
+the command being run is a KV command, which is evaluated and then
+replicated to Raft followers.
+
+- An errant node has gone dark and is considered non-live.
+- The minimum required version has been upped, but the gossiped
+  `ClusterVersion` has not yet been received by all nodes.
+- The non-live node suddenly re-surfaces.
+- The new feature is evaluated on a node which has received the
+  incremented minimum required version.
+- It attempts to send to the node that has just re-surfaced, but
+  receives an error indicating it should backoff and retry while
+  the errant node waits to receive the updated `ClusterVersion`.
+
+Similarly, consider another scenario:
+
+- An errant node has gone dark and is considered non-live.
+- **The cluster has been upgraded, but the errant node is unreachable
+  and is skipped.**
+- The minimum required version has been upped, but the gossiped
+  `ClusterVersion` has not yet been received by all nodes.
+- The non-live node suddenly re-surfaces.
+- The new feature is evaluated on a node which has received the
+  incremented minimum required version.
+- It attempts to send to the node that has just re-surfaced, but
+  receives an error indicating the receiver is out of date and it
+  should backoff and retry. In this case, the receiver will commit
+  suicide.
+
+## SQL syntax
+
+The operator can upgrade the cluster's version in order to use new
+features available at that version via the `SET CLUSTER VERSION
```

Comment:
Agreed with alex here we should recycle the existing feature `set cluster setting` instead of inventing yet one new set/show syntax format.